pages/[user].tsx

```diff
@@ -89,51 +105,36 @@ export default function User(props: inferSSRProps<typeof getServerSideProps>) {
     </>
   );
 }
-
-export const getServerSideProps = async (context: GetServerSidePropsContext) => {
-  const username = (context.query.user as string).toLowerCase();
-
-  const user = await prisma.user.findUnique({
-    where: {
-      username,
-    },
+export const getStaticPaths: GetStaticPaths = async () => {
```

Comment:
This makes sure that we prerender all the users in the db. Will become heavy once we have tens of thousands and then there's other approaches to this as well.

Reply:
What other approaches do you recommend?

Reply:
What they refer to as "Incremental Static Regeneration". Essentially skipping `getStaticPaths` or reducing it to only build active users.

> For example, you can generate the most popular 1,000 products at build-time by returning the paths for the top 1,000 product IDs in getStaticPaths.

https://vercel.com/docs/next.js/incremental-static-regeneration?query=getstaticp#generating-paths